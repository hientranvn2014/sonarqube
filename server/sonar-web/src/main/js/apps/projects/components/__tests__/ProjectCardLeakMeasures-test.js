/*
 * SonarQube
 * Copyright (C) 2009-2017 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import React from 'react';
import { shallow } from 'enzyme';
import ProjectCardLeakMeasures from '../ProjectCardLeakMeasures';

const measures = {
  alert_status: 'ERROR',
  new_reliability_rating: '1.0',
  new_bugs: '8',
  new_security_rating: '2.0',
  new_vulnerabilities: '2',
  new_maintainability_rating: '1.0',
  new_code_smells: '0',
  new_coverage: '26.55',
  new_duplicated_lines_density: '0.55',
  new_lines: '87'
};

it('should render correctly with all data', () => {
  const wrapper = shallow(<ProjectCardLeakMeasures measures={measures} />);
  expect(wrapper).toMatchSnapshot();
});

it('should not render new coverage', () => {
  const wrapper = shallow(
    <ProjectCardLeakMeasures measures={{ ...measures, new_coverage: undefined }} />
  );
  expect(wrapper.find('[data-key="new_coverage"]')).toMatchSnapshot();
});

it('should not render new duplications', () => {
  const wrapper = shallow(
    <ProjectCardLeakMeasures measures={{ ...measures, new_duplicated_lines_density: undefined }} />
  );
  expect(wrapper.find('[data-key="new_duplicated_lines_density"]')).toMatchSnapshot();
});

it('should not render new lines', () => {
  const wrapper = shallow(
    <ProjectCardLeakMeasures measures={{ ...measures, new_lines: undefined }} />
  );
  expect(wrapper.find('[data-key="new_lines"]')).toMatchSnapshot();
});
