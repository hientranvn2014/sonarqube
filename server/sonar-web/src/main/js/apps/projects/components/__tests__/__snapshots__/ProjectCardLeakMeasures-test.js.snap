// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render new coverage 1`] = `
<div
  className="project-card-measure"
  data-key="new_coverage"
>
  <div
    className="project-card-measure-inner"
  >
    <div
      className="project-card-measure-number"
    >
      <Measure
        measure={
          Object {
            "leak": undefined,
          }
        }
        metric={
          Object {
            "key": "new_coverage",
            "type": "PERCENT",
          }
        }
      />
    </div>
    <div
      className="project-card-measure-label"
    >
      metric.new_coverage.name
    </div>
  </div>
</div>
`;

exports[`should not render new duplications 1`] = `
<div
  className="project-card-measure"
  data-key="new_duplicated_lines_density"
>
  <div
    className="project-card-measure-inner"
  >
    <div
      className="project-card-measure-number"
    >
      <Measure
        measure={
          Object {
            "leak": undefined,
          }
        }
        metric={
          Object {
            "key": "new_duplicated_lines_density",
            "type": "PERCENT",
          }
        }
      />
    </div>
    <div
      className="project-card-measure-label"
    >
      metric.new_duplicated_lines_density.short_name
    </div>
  </div>
</div>
`;

exports[`should not render new lines 1`] = `undefined`;

exports[`should render correctly with all data 1`] = `
<div
  className="project-card-leak-measures"
>
  <div
    className="project-card-measure smaller-card"
    data-key="new_reliability_rating"
  >
    <div
      className="project-card-measure-inner"
    >
      <div
        className="project-card-measure-number"
      >
        <Measure
          className="spacer-right"
          measure={
            Object {
              "leak": "8",
            }
          }
          metric={
            Object {
              "key": "new_bugs",
              "type": "SHORT_INT",
            }
          }
        />
        <Rating
          muted={false}
          small={false}
          value="1.0"
        />
      </div>
      <div
        className="project-card-measure-label-with-icon"
      >
        <BugIcon
          className="little-spacer-right vertical-bottom"
        />
        metric.new_bugs.name
      </div>
    </div>
  </div>
  <div
    className="project-card-measure"
    data-key="new_security_rating"
  >
    <div
      className="project-card-measure-inner"
    >
      <div
        className="project-card-measure-number"
      >
        <Measure
          className="spacer-right"
          measure={
            Object {
              "leak": "2",
            }
          }
          metric={
            Object {
              "key": "new_vulnerabilities",
              "type": "SHORT_INT",
            }
          }
        />
        <Rating
          muted={false}
          small={false}
          value="2.0"
        />
      </div>
      <div
        className="project-card-measure-label-with-icon"
      >
        <VulnerabilityIcon
          className="little-spacer-right vertical-bottom"
        />
        metric.new_vulnerabilities.name
      </div>
    </div>
  </div>
  <div
    className="project-card-measure"
    data-key="new_maintainability_rating"
  >
    <div
      className="project-card-measure-inner"
    >
      <div
        className="project-card-measure-number"
      >
        <Measure
          className="spacer-right"
          measure={
            Object {
              "leak": "0",
            }
          }
          metric={
            Object {
              "key": "new_code_smells",
              "type": "SHORT_INT",
            }
          }
        />
        <Rating
          muted={false}
          small={false}
          value="1.0"
        />
      </div>
      <div
        className="project-card-measure-label-with-icon"
      >
        <CodeSmellIcon
          className="little-spacer-right vertical-bottom"
        />
        metric.new_code_smells.name
      </div>
    </div>
  </div>
  <div
    className="project-card-measure"
    data-key="new_coverage"
  >
    <div
      className="project-card-measure-inner"
    >
      <div
        className="project-card-measure-number"
      >
        <Measure
          measure={
            Object {
              "leak": "26.55",
            }
          }
          metric={
            Object {
              "key": "new_coverage",
              "type": "PERCENT",
            }
          }
        />
      </div>
      <div
        className="project-card-measure-label"
      >
        metric.new_coverage.name
      </div>
    </div>
  </div>
  <div
    className="project-card-measure"
    data-key="new_duplicated_lines_density"
  >
    <div
      className="project-card-measure-inner"
    >
      <div
        className="project-card-measure-number"
      >
        <Measure
          measure={
            Object {
              "leak": "0.55",
            }
          }
          metric={
            Object {
              "key": "new_duplicated_lines_density",
              "type": "PERCENT",
            }
          }
        />
      </div>
      <div
        className="project-card-measure-label"
      >
        metric.new_duplicated_lines_density.short_name
      </div>
    </div>
  </div>
  <div
    className="project-card-measure smaller-card pull-right"
    data-key="new_lines"
  >
    <div
      className="project-card-measure-inner"
    >
      <div
        className="project-card-measure-number"
      >
        <Measure
          measure={
            Object {
              "leak": "87",
            }
          }
          metric={
            Object {
              "key": "new_lines",
              "type": "SHORT_INT",
            }
          }
        />
      </div>
      <div
        className="project-card-measure-label"
      >
        metric.new_lines.short_name
      </div>
    </div>
  </div>
</div>
`;
